content/post/text-analysis-of-the-10th-republican-presidential-candidate-debate-using-r-and-the-quanteda-package.Rmd

```diff
@@ -0,0 +1,222 @@
+
+In both examples, we can see that Trump spoke the most and Carson the least.
+
+## Analysis: What were they saying?
+
+If we wanted to go further, we convert the segmented corpus into a _document-feature matrix_ and apply one of many available psychological dictionaries to analyze the tone of each candidate’s remarks.  
+
+Here we demonstrate this using the Regressive Imagery Dictionary, from Martindale, C. (1975) _Romantic Progression: The Psychology of Literary History._ Washington, D.C.: Hemisphere. The code below automatically downloads a version of this dictionary in a format prepared for the WordStat software by [Provalis](http://www.provalisresearch.com), available from http://www.provalisresearch.com/Download/RID.ZIP. **quanteda** can import dictionaries formatted for WordStat, using the `dictionary()` function.  We will apply the RID dictionary to find out who used what degree of "glory"-oriented language. (You might be able to guess the results already.)
+
+```{r}
+# get the RID from the Provalis website
+download.file("http://provalisresearch.com/Download/RID.ZIP", "RID.zip")
+unzip("RID.zip")
+data_dictionary_RID <- dictionary(file = "RID.CAT", format = "wordstat")
+invisible(file.remove("RID.zip", "RID.CAT", "RID.exc"))
+```
+
+This is a nested dictionary object with three primary keys:
+```{r}
+names(data_dictionary_RID)
+```
+
+There are additional keys nested inside each of these.  We can show the number of values for each nested entry for the "EMOTIONS" top-level key.
+```{r}
+lengths(data_dictionary_RID[["EMOTIONS"]])
+```
+
+We can inspect the category we will use ("Glory") by looking at the last sub-key of the third key, "Emotions".
+```{r}
+tail(data_dictionary_RID[["EMOTIONS"]], 1)
+```
+
+Let's create a document-feature matrix from the candidate corpus, grouping the documents by speaker.  This takes all of the speech acts and combines them by the value of "speaker", so that the new number of documents is just five (one for each candidate).
+```{r}
+dfmatcand <- dfm(corpcand, groups = "speaker", verbose = TRUE)
+```
+
+Because the texts are of different lengths, we want to normalize them (by converting the feature counts into vectors of relative frequencies within document):
+```{r}
+dfmatcand <- dfm_weight(dfmatcand, "prop")
+```
+
+Now we are in a position to apply the RID to the dfm, which matches on the "glob" formatted wildcard expressions that form the values of the RID in our `data_dictionary_RID` object.
+```{r}
+dfmatcandRID <- dfm_lookup(dfmatcand, dictionary = data_dictionary_RID)
```

Comment:
I made a small change and pushed it to the branch to avoid a warning message: 

```r
head(dfmatcandRID, nf = 4) %>%
        convert(to = "data.frame") %>% 
        knitr::kable()
```

Without `convert(to = "data.frame") the dfm is transformed using `as.data.frame()`, which results in a warning:

```
(Warning message:
'as.data.frame.dfm' is deprecated.
Use 'convert(x, to = "data.frame")' instead.
See help("Deprecated")`